src/audio/host-zephyr.c

```diff
@@ -302,17 +303,11 @@ void host_common_update(struct host_data *hd, struct comp_dev *dev, uint32_t byt
 			/* send timestamped position to host
 			 * (updates position first, by calling ops.position())
 			 */
-			update_mailbox = true;
-			send_ipc = true;
-		}
-	}
-
-	if (update_mailbox) {
-		pipeline_get_timestamp(dev->pipeline, dev, &hd->posn);
-		mailbox_stream_write(dev->pipeline->posn_offset,
-				     &hd->posn, sizeof(hd->posn));
-		if (send_ipc)
+			pipeline_get_timestamp(dev->pipeline, dev, &hd->posn);
```

Comment:
can it be that we arrive here with `hd->cont_update_posn == true`? Then we would call `pipeline_get_timestamp()` and `mailbox_stream_write()` twice?

Reply:
that's really tricky case, in theory, it is possible....
they are both assigned by params, if driver pass both as true, then this is the case.